src/ophyd_async/epics/pmac/_utils.py

```diff
@@ -104,6 +105,28 @@ def from_slice(cls, slice: Slice[Motor], ramp_up_time: float) -> _Trajectory:
             durations=durations,
         )
 
+    def with_ramp_down(
+        self,
+        ramp_down_pos: dict[Motor, np.float64],
+        ramp_down_time: float,
+        ramp_down_velocity: float,
+    ) -> _Trajectory:
+        new_time = np.append(self.durations, [int(ramp_down_time / TICK_S)])
+        new_pos = {
+            motor: np.append(self.positions[motor], [ramp_down_pos[motor]])
+            for motor in ramp_down_pos.keys()
+        }
+        new_vel = {
+            motor: np.append(self.velocities[motor], [ramp_down_velocity])
+            for motor in ramp_down_pos.keys()
+        }
+        return _Trajectory(
+            positions=new_pos,
+            velocities=new_vel,
+            user_programs=self.user_programs,
+            durations=new_time,
+        )
```

Comment:
Should: Since this is an instance method, and we are mutating the instance variables with the new ramp down info, we could just return self:

```suggestion
        return self
```

Reply:
That makes sense! And the code looks nicer with your suggestion